### PR TITLE
Tank Dispenser Refactor

### LINKED ISSUE #15494: Re-inserting tanks in a tank storage unit puts them at the back, should be last-in-first-out
#### Issue bingo

Please check whatever applies. More checkboxes checked increase your chances of not being yelled at by every contributor.

- [ ] Issue could be reproduced at least once
- [ ] Issue could be reproduced by different players
- [ ] Issue could be reproduced in multiple rounds
- [ ] Issue happened in a recent (less than 7 days ago) round
- [ ] [Couldn't find an existing issue about this](https://github.com/d3athrow/vgstation13/issues)

### PR DESCRIPTION
**Pull request by Kurfursten**
Previous functionality:
* Tanks in tank dispensers are spawned on demand
* Prioritizes creating new tank over removing existing tank
* FIFO after all tanks are spawned

New functionality
* Tanks are spawned at roundstart
* LIFO

Fully tested
fixes #15494

🆑 
* tweak: Tank dispensers are now LIFO (were FIFO)
* tweak: Tank dispensers will now attempt to place removed tanks in your hand instead of the floor

### CODE REVIEW

**Comment by Exxion** [comment]
I think your changelog is backwards
They were FIFO, now they're FILO

**Comment by Kurfursten** [comment]
Yeah you're right, I actually changed it seconds before you hit send

**Comment by Exxion** [comment]
To quote the forum traditions of old, I apparently just got ninja'd

File: code/game/objects/structures/tank_dispenser.dm [hunk 1]
```diff
@@ -24,19 +24,26 @@
 
 /obj/structure/dispenser/New()
 	. = ..()
+	while(oxygentanks>0)
```
Comment (PJB3005) [inline code/game/objects/structures/tank_dispenser.dm:27]: Can't you make tank dispensers with like plasteel or something? What happens to those?

Also, use a for loop.

**Review by PJB3005 — COMMENTED** [review]
Why exactly make them immediately initialized? What's the advantage other than (maybe) reducing code complexity.

File: code/game/objects/structures/tank_dispenser.dm (hunk 1 above)
Comment (Kurfursten) [reply code/game/objects/structures/tank_dispenser.dm:27]: They have a value of oxygentanks = 0 (you can see this at the top of this file)

And of course I tested it just to be sure, and to check icon states and stuff

**Comment by Jalleo** [comment]
@PJB3005 if its currently the same code for basically every other codebase you can throw in used tanks for half full tanks.

**Comment by PJB3005** [comment]
Uh, no you can't.

**Comment by Kurfursten** [comment]
?

**Comment by PJB3005** [comment]
The guy I was responding to deleted his comment. I still have a back up in my email if you really want it.

File: code/game/objects/structures/tank_dispenser.dm (hunk 1 above)
Comment (N3X15) [reply code/game/objects/structures/tank_dispenser.dm:27]: If this causes an infinite loop, I'm not going to be happy.

**Pull request merged**